packages/rs-json-schema-compatibility-validator/src/rules/rule_set.rs

```diff
@@ -0,0 +1,1378 @@
+static U64_BIGGER_CALLBACK: Lazy<IsReplacementAllowedCallback> = Lazy::new(|| {
+    Some(Arc::new(|schema, op| {
+        let original_value = schema.try_pointer(&op.path)?.try_to_u64()?;
+        let new_value = op.value.try_to_u64()?;
+
+        Ok(original_value < new_value)
+    }))
+});
+
+/// Replacement is allowed if a new u64 value is smaller than original
+static U64_SMALLER_CALLBACK: Lazy<IsReplacementAllowedCallback> = Lazy::new(|| {
+    Some(Arc::new(|schema, op| {
+        let original_value = schema.try_pointer(&op.path)?.try_to_u64()?;
+        let new_value = op.value.try_to_u64()?;
+
+        Ok(original_value > new_value)
+    }))
+});
+
+/// Replacement is allowed if a new f64 value is bigger than original
+static F64_BIGGER_CALLBACK: Lazy<IsReplacementAllowedCallback> = Lazy::new(|| {
+    Some(Arc::new(|schema, op| {
+        let original_value = schema.try_pointer(&op.path)?.try_to_f64()?;
+        let new_value = op.value.try_to_f64()?;
+
+        Ok(original_value < new_value)
+    }))
+});
+
+/// Replacement is allowed if a new f64 value is smaller than original
+static F64_SMALLER_CALLBACK: Lazy<IsReplacementAllowedCallback> = Lazy::new(|| {
+    Some(Arc::new(|schema, op| {
+        let original_value = schema.try_pointer(&op.path)?.try_to_f64()?;
+        let new_value = op.value.try_to_f64()?;
+
+        Ok(original_value > new_value)
+    }))
+});
+
+/// Replacement is allowed if a new value is an existing element in the original array
+static EXISTING_ELEMENT_CALLBACK: Lazy<IsReplacementAllowedCallback> = Lazy::new(|| {
+    Some(Arc::new(|schema, op| {
+        // One segment back to required array
+        let path = PathBuf::from(&op.path);
+        let required_path = path.parent().and_then(|p| p.to_str()).ok_or_else(|| {
+            InvalidJsonPatchOperationPathError {
+                path: op.path.clone(),
+            }
+        })?;
+
+        let original_required_value = schema.try_pointer(required_path)?;
+
+        let original_required_array_of_values =
+            original_required_value.as_array().ok_or_else(|| {
+                Error::UnexpectedJsonValueType(UnexpectedJsonValueTypeError {
+                    expected_type: "array".to_string(),
+                    value: original_required_value.clone(),
+                })
+            })?;
+
+        Ok(original_required_array_of_values.contains(&op.value))
+    }))
+});
+
+/// The rules define, which change in JSON Schema for a keyword and its inner structure is compatible or not.
+/// Important note: Not all keywords are supported, and some rules are
+/// implemented based on Data Contract schema validation requirements
+pub static KEYWORD_COMPATIBILITY_RULES: Lazy<HashMap<&'static str, CompatibilityRules>> =
+    Lazy::new(|| {
+        HashMap::from_iter([
+            (
+                "$id",
+                CompatibilityRules {
+                    allow_addition: true,
```

Comment:
why can we allow the addition?

Reply:
This is just an anchor for `$ref`. You can add anchors to reuse existing validation logic for a new schema elements, but you can add `$ref` to change validation logic of existing elements.